fontdrasil/src/orchestration.rs

```diff
@@ -4,15 +4,22 @@ use std::{
     collections::HashSet,
     fmt::{Debug, Display},
     hash::Hash,
-    sync::Arc,
 };
 
 pub const MISSING_DATA: &str = "Missing data, dependency management failed us?";
 
+/// Identifies all items of the same group, typically enum variant, of an identifer
+///
+/// String because it's enormously easier to understand debug output that way
+pub type IdentifierDiscriminant = &'static str;
+
 /// A type that affords identity.
 ///
 /// Frequently copied, used in hashmap/sets and printed to logs, hence the trait list.
-pub trait Identifier: Debug + Clone + Eq + Hash {}
+pub trait Identifier: Debug + Clone + Eq + Hash {
+    /// Return a value that is consistent across all instances in the same group, e.g. enum variant.
+    fn discriminant(&self) -> IdentifierDiscriminant;
```

Comment:
also: what is this used for? You might be able to just use [`type_id`](https://doc.rust-lang.org/1.74.0/std/any/trait.Any.html#tymethod.type_id) or [`type_name`](https://doc.rust-lang.org/1.74.0/std/any/fn.type_name.html)?

Reply:
The intent is identifiers are usually enums; you want to express all of a variant so there isn't a type.

Reply:
Given the lack of a type for a variant I think this is as intended